--- a/configure
+++ b/configure
@@ -16012,7 +16012,7 @@
 
 
 if test "$OPENLSWS_BSSL" = no ; then
-    if test "$OSNAME" = Darwin ; then
+    if test "$OSNAME" = Linux ; then
         usedynossl=yes
         CPPFLAGS="$CPPFLAGS -I../../src -I../../include -I/usr/local/include -I/usr/local -I/usr -I/usr/pkg "
     else
@@ -16490,7 +16490,7 @@
 
 
 
-if test "$OSNAME" = Darwin ; then
+if test "$OSNAME" = Linux ; then
 
 
 # Check whether --with-openssl was given.
